apps/test/solutions/applab/cspunit3.js

```diff
@@ -0,0 +1,179 @@
+  "showTurtleBeforeRun": true,
+  "autocompletePaletteApisOnly": true,
+  "textModeAtStart": false,
+  "designModeAtStart": false,
+  "hideDesignMode": true,
+  "beginnerMode": false,
+  "markdownInstructions": "<img src=\"https://images.code.org/ad48e7224312a6c41f4fc5727af53cc0-image-1436287265071.png\" align=right> **Warm up 2:** Draw a 1 x 1 square to the front and right of the turtle as efficiently as possible.  The program should stop with turtle in its original position, facing its original direction.\r\n\r\nWhen you're done click the Finish button to move onto the next problem.\r\n\r\n",
+  "puzzle_number": 3,
+  "stage_total": 7,
+  "noPadding": null,
+  "lastAttempt": "",
+  "levelHtml": "<div xmlns=\"http://www.w3.org/1999/xhtml\" id=\"divApplab\" class=\"appModern\" tabindex=\"1\" style=\"width: 200px; height: 200px;\"><div class=\"screen\" tabindex=\"1\" id=\"screen1\" style=\"display: block; height: 200px; width: 200px; left: 0px; top: 0px; position: absolute; z-index: 0;\"></div></div>",
+  "id": "custom",
+};
+
+
+
+/**
+ * Simulates dragging the nth block in the toolbox into the canvas at line
+ * targetIndex (assumes all blocks are size 30).
+ * @param {number} blockIndex Nth block in toolbox
+ * @param {number} targetIndex Nth line in target
+ */
+function dragToolboxBlock(blockIndex, targetIndex) {
+  var start = {
+    x: $(".droplet-palette-canvas").eq(0).offset().left + 10,
+    y: $(".droplet-palette-canvas").eq(0).offset().top + 10 +
+      blockIndex * 30
+  };
+  var end = {
+    x: $(".droplet-main-canvas").eq(0).offset().left,
+    y: $(".droplet-main-canvas").eq(0).offset().top +
+      targetIndex * 30
+  };
+
+  var mousedown = testUtils.createMouseEvent('mousedown', start.x, start.y);
+  var drag = testUtils.createMouseEvent('mousemove', end.x, end.y);
+  var mouseup = testUtils.createMouseEvent('mouseup', end.x, end.y);
+
+  $(".droplet-drag-cover")[0].dispatchEvent(mousedown);
+  $(".droplet-drag-cover")[0].dispatchEvent(drag);
+  $(".droplet-drag-cover")[0].dispatchEvent(mouseup);
+}
+
+// Extract a list of those pixels that are actually filled
+function getColoredPixels(imageData, width, height) {
+  var list = [];
+  for (var y = 0; y < 400; y++) {
+    for (var x = 0; x < 400; x++) {
+      // we only care about the alpha bit
+      var index = (y * 400 + x) * 4 + 3;
+      if (imageData.data[index] !== 0) {
+        list.push([x, y]);
+      }
+    }
+  }
+  return list;
+}
+
+module.exports = {
+  app: "applab",
+  skinId: "applab",
+  levelDefinition: levelDefinition,
+  tests: [
+    {
+      description: "Expected solution.",
+      editCode: true,
+      xml: '',
+      runBeforeClick: function (assert) {
+        var moveForward = 0;
+        var turnLeft = 1;
+
+        var nextIndex = 0;
+        function appendToolboxBlock(blockIndex) {
+          dragToolboxBlock(blockIndex, nextIndex++);
+        }
+
+        appendToolboxBlock(moveForward);
```

Comment:
what default parameter are you assuming for the moveForward block in the block tray? It would be helpful to note this assumption in comments, so if it somehow changes then we have a hint as to why this breaks.